src/server/coprocessor/endpoint.rs

```diff
@@ -222,19 +239,17 @@ impl BatchRunnable<Task> for Host {
                         }
                     };
                     let len = reqs.len() as f64;
-                    let running_count = self.running_count.clone();
-                    if running_count.load(Ordering::SeqCst) >= self.max_running_task_count {
+
+                    if self.pool.get_task_count() >= self.max_running_task_count {
                         notify_batch_failed(Error::Full(self.max_running_task_count), reqs);
                         continue;
                     }
-                    running_count.fetch_add(reqs.len(), Ordering::SeqCst);
                     COPR_PENDING_REQS.with_label_values(&["select"]).add(len);
                     for req in reqs {
-                        let running_count = running_count.clone();
                         let end_point = TiDbEndPoint::new(snap.clone());
-                        self.pool.execute(move || {
+                        let txn_id = req.start_ts.unwrap_or_default();
+                        self.pool.execute(txn_id, move || {
```

Comment:
It would be changed to much here If we check error before push the task into `threadpool`, maybe we could do in another PR. @hhkbp2